app/controllers/activities_controller.rb

```diff
@@ -0,0 +1,36 @@
+class ActivitiesController < ApplicationController
+	include ActivitiesHelper
+	before_filter :redirect_to_root_if_not_logged_in
+	before_filter :redirect_to_root_if_not_admin, only: :dashboard
+	before_action :set_todays_activity, except: :dashboard
+
+	def index
+		@activities = current_user.get_authorized_activities(user_id: params['user'])
+	end
+
+	def edit
+	end
+
+	def update
+		@todays_activity = current_user.todays_activity
+		updated_activity = convert_to_24_hours(params_activity)
```

Comment:
I assume this is the method you are calling from helper.
If this is heavily UI related, then it should be called in view or helper; not controller.